rtl8195a: add first port of rtl8195a

cwyark opened the pull request:
Rtl8195a is a WiFi soc with ARM cortex M3, 2MB sdram and 1MB flash. The reference board is called Ameba. Currently, REPL, GPIO and WiFi is workable and the build environment are OSX and Windows only.

cwyark committed:
rt8195a: add rt8195a initial files

add rt8195a source files and Makefile

cwyark committed:
rt8195a: add rt8195a initial files

add rt8195a source files and Makefile

cwyark committed:
rtl8195a: add Makeifle and essential source files

cwyark committed:
rtl8195a: add Makeifle and essential source files

cwyark committed:
Merge remote-tracking branch 'upstream/master' into rt8195-dev

cwyark committed:
Merge branch 'master' of github.com:cwyark/micropython

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
rtl8195a: update Makefile and ameba's sdk and library

cwyark committed:
rtl8195a: update Makefile and ameba's sdk and library

cwyark committed:
rtl8195a: add image create procedure in Makefile

cwyark committed:
rtl8195a: add image create procedure in Makefile

cwyark committed:
rtl8195a: update shared library

cwyark committed:
rtl8195a: update shared library

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
rt8195a: dos2unix the linker file

cwyark committed:
rt8195a: update dll files

cwyark committed:
rt8195a: update dll files

cwyark committed:
Merge remote-tracking branch 'origin/master' into rt8195-dev

cwyark committed:
rt8195a: update dll files

cwyark committed:
rt8195a: update dll files

cwyark committed:
rt8195a: delete lib*.a

cwyark committed:
rt8195a: delete lib*.a

cwyark committed:
rt8195a: update linked libraries

cwyark committed:
rt8195a: update linked libraries

cwyark committed:
rt8195a: rename lib folder name to linklibs

cwyark committed:
rt8195a: rename lib folder name to linklibs

cwyark committed:
rt8195a: update libameba.a

cwyark committed:
rt8195a: update lib_ameba.a

cwyark committed:
rt8195a: update lib_ameba.a`

cwyark committed:
rt8195: remove *.a

cwyark committed:
rt8195a: add *.a

cwyark committed:
rt8195a: add tools for mac

cwyark committed:
rt8195a: add tools for mac

cwyark committed:
rt8195a: first porting to the raw ane pure micropython REPL

cwyark committed:
rt8195a: first porting to the raw ane pure micropython REPL

cwyark committed:
Merge remote-tracking branch 'upstream/master' into rt8195-dev

cwyark committed:
rt8195a: add exceptions and hardware module

Add Pin type and it is arranged to hardware module

cwyark committed:
rt8195a: add exceptions and hardware module

Add Pin type and it is arranged to hardware module

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' into rt8195-dev

cwyark committed:
rtl8195a: add pin mapping files

use pin mapping method as same as cc3200 and stm series board
add ameba board pin definations

cwyark committed:
rtl8195a: add pin mapping files

use pin mapping method as same as cc3200 and stm series board
add ameba board pin definations

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' into rt8195-dev

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' of github.com:cwyark/micropython

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' into rt8195-dev

cwyark committed:
.gitignore: merge gitignore from foxhunt-dev branch

cwyark committed:
.gitignore: merge gitignore from foxhunt-dev branch

cwyark committed:
rtl8195a: update library from Ameba sdk

cwyark committed:
rtl8195a: update Ameba sdk headers

cwyark committed:
rtl8195a: update Ameba sdk headers

cwyark committed:
rtl8195a: add wifi object in wireless module

cwyark committed:
rtl8195a: add wifi object in wireless module

cwyark committed:
rtl8195a: add os task for main function and the wifi scanning command

cwyark committed:
rtl8195a: add os task for main function and the wifi scanning command

cwyark committed:
rtl8195a: complete wifi WEP auth method

cwyark committed:
rtl8195a: complete wifi WEP auth method

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' of github.com:cwyark/micropython

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' of github.com:cwyark/micropython

cwyark committed:
Merge branch 'master' into rt8195-dev

cwyark committed:
rtl8195a: rename rt8195a to rtl8195a

cwyark committed:
rtl8195a: update bsp pre-build firmware

cwyark committed:
rtl8195a: update bsp pre-build firmware

dpgeorge commented:
Thanks for the contribution.  Unfortunately this is too much code to merge in one go, if at all.  There is a lot of vendor-specific code here, and some duplicated stuff (why multiple versions of lwip?), and lwip 1.4.1 already exists in our repository so should be used instead of including it again.

The Ameba dev board does not look like it's easily available.  And if we can't compile this port under Linux then it's hard to test.

What are your long term plans for this port?

cwyark commented:
Thanks for your information and sorry for the mess commits.Ameba board is quit cheap and easy to get here (Taiwan). Mostly we use this board to DIY some air pollution monitoring or something we need to place the board outside the door. Micropython is really fit for such applications that we can remotely control or debug. 
My long term plans for RTL8195A porting is to make it easy to develop application like Wipy. For example, you can use ftp to upgrade your .py script and use telnet to remotely control your board. 
So here's my road map for this port:
1. gpio/ spi / i2c  
2. WiFi and NFC functions 
3. networking ( sockets ).
4. OTA using FTP / HTTP.

I know there are lots of vendor-specific code even I need to link their pre-build libraries. The .elf file can be build under linux (arm-none-eabi-gcc), but the final binary generating is processed under a program and the program can only be executed under linux / osx. These things can be solved because the vendor seems released their source somewhere in github but few people know. 
What is your suggestions or something need to be improved if I can contribute this port ?

cwyark committed:
rtl8195a: update lib_ameba.a

cwyark committed:
rtl8195a: update lib_ameba.a

cwyark committed:
rtl8195a: update *.dll and linker script

cwyark committed:
Merge branch 'master' of github.com:cwyark/micropython

cwyark committed:
Saving files before refreshing line endings

cwyark committed:
rtl8195a: remove rt8195a folder

cwyark committed:
Saving files before refreshing line endings

cwyark committed:
Merge branch 'rt8195-dev' of github.com:cwyark/micropython into rt8195-dev

cwyark committed:
rtl8195a: remote unused file .version

cwyark committed:
rtl8195a: fix compile order issue for the pins.c

pins.c will be compiled after linkinng, and this will cause link failed

cwyark committed:
rtl8195a: enable REPL auto indent in readline

cwyark committed:
rtl8195a: copy modlwip from extmode/modlwip.c and enable lwip module

cwyark committed:
rtl8195a: add wifi features

1. rename wifi object to WIFI
2. add WEP auth method
3. remove lwip_init in wifi module

cwyark committed:
rtl8195a: enlarge heap and stack size for micropython

cwyark committed:
Merge branch 'master' of https://github.com/micropython/micropython

cwyark committed:
rtl8195a: add pre build image under tools/prebuild

cwyark committed:
rtl9105a: update ram_all.bin

cwyark committed:
rtl8195a: remove prebuild in rtl8195a branch

cwyark committed:
rtl9195a: remove ram_all.bin prebuild and replaced with a ziped file

It will changed the codec in the bin file and make it unuseable

cwyark committed:
rtl8195a: change ram_all.zip to ram_all.tar

zip compression will destroy the bin file after uncompress, so use tar

cwyark committed:
rtl8195a: update ram_all.tar for pre release

cwyark committed:
rtl8195a: add I2C proto type functions

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'master' into rtl8195a-dev

cwyark committed:
rtl8195a: add i2c hal call functions

cwyark committed:
rtl815a: remove tools/ram_all.tar

this file will be different when you commit and checkout, so I'll
release the fw in my space

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
Merge branch 'rtl8195a-dev'

cwyark committed:
Merge branch 'master' of https://github.com/micropython/micropython

cwyark committed:
Merge remote-tracking branch 'upstream/master'

cwyark committed:
gitigrnoe: update gitignore for .ropeproject

cwyark committed:
Merge branch 'master' of https://github.com/micropython/micropython

cwyark commented:
RTL8105A ports will be keep in my repository. There are too many vendor's code and closed source libraries.
